.github/workflows/linux-build-clang.yaml

```diff
@@ -23,6 +23,9 @@ jobs:
     steps:
       - name: Checkout Repository
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: true
```

Comment:
doing a full checkout adds ~10s to this step comparing to older runs.  

I think we should add it to it's own step, ie. `Checkout Repository with Tags` that has an `if: ${{ inputs.uploadArtifacts }}` and then this step can have an `if: !${{ inputs.uploadArtifacts }}`.  This is so it only runs when it is required on a release workflow.